config.go

```diff
@@ -26,6 +27,24 @@ type defaultsHandler interface {
 	SetDefaults(c *Config) error
 }
 
+// Define backoffStrategy
+type BackoffStrategy interface {
+   Calculate(attempt int) time.Duration
+}
+
+type ExponentialStrategy struct {
+	config *Config
```

Comment:
if `config` isn't exported, you won't actually be able to instantiate it outside this package.